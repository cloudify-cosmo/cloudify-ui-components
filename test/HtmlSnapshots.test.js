import initStoryshots from '@storybook/addon-storyshots';
import ReactDOM from 'react-dom';
import _ from 'lodash';

ReactDOM.createPortal = _.identity;
ReactDOM.findDOMNode = () => ({ addEventListener: _.noop });
initStoryshots();
