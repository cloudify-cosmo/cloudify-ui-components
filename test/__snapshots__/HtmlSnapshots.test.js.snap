// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Buttons/ApproveButton Basic 1`] = `
<button
  className="ui button ok"
  onClick={[Function]}
>
  <i
    aria-hidden="true"
    className="checkmark icon"
    onClick={[Function]}
  />
  Save
</button>
`;

exports[`Storyshots Buttons/ApproveButton Custom 1`] = `
<button
  className="ui green button ok"
  onClick={[Function]}
>
  <i
    aria-hidden="true"
    className="add user icon"
    onClick={[Function]}
  />
  Add
</button>
`;

exports[`Storyshots Buttons/CancelButton Basic 1`] = `
<button
  className="ui button basic cancel"
  onClick={[Function]}
>
  <i
    aria-hidden="true"
    className="remove icon"
    onClick={[Function]}
  />
  Cancel
</button>
`;

exports[`Storyshots Buttons/CancelButton Custom 1`] = `
<button
  className="ui red button basic cancel"
  onClick={[Function]}
>
  <i
    aria-hidden="true"
    className="stop icon"
    onClick={[Function]}
  />
  Stop
</button>
`;

exports[`Storyshots Buttons/CopyToClipboardButton Basic 1`] = `
<button
  className="ui basic compact icon button"
  onClick={[Function]}
>
  <i
    aria-hidden="true"
    className="copy icon"
    onClick={[Function]}
  />
</button>
`;

exports[`Storyshots Buttons/CopyToClipboardButton Custom 1`] = `
<button
  className="ui basic vertical animated button"
  onClick={[Function]}
>
  <div
    className="visible content"
  >
    Copy ID
  </div>
  <div
    className="hidden content"
  >
    <i
      aria-hidden="true"
      className="copy icon"
      onClick={[Function]}
    />
  </div>
</button>
`;

exports[`Storyshots Elements/Checkmark Basic 1`] = `
<i
  aria-hidden="true"
  className="square outline icon"
  onClick={[Function]}
  title="No"
/>
`;

exports[`Storyshots Elements/Checkmark Checked 1`] = `
<i
  aria-hidden="true"
  className="checkmark box icon"
  onClick={[Function]}
  title="Yes"
/>
`;

exports[`Storyshots Elements/Checkmark Unchecked 1`] = `
<i
  aria-hidden="true"
  className="square outline icon"
  onClick={[Function]}
  title="No"
/>
`;

exports[`Storyshots Elements/Dropdown Basic 1`] = `
<div
  style={
    Object {
      "height": 200,
    }
  }
>
  <div
    aria-expanded={false}
    className="ui search selection dropdown"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="combobox"
  >
    <input
      aria-autocomplete="list"
      autoComplete="off"
      className="search"
      onChange={[Function]}
      style={
        Object {
          "width": undefined,
        }
      }
      tabIndex={0}
      type="text"
      value=""
    />
    <div
      aria-atomic={true}
      aria-live="polite"
      className="text"
      role="alert"
    >
      Option 1
    </div>
    <i
      aria-hidden="true"
      className="dropdown icon clear"
      onClick={[Function]}
    />
    <div
      className="menu transition"
      role="listbox"
    >
      <div
        aria-checked={true}
        aria-selected={true}
        className="active selected item"
        onClick={[Function]}
        option-value="option1"
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          Option 1
        </span>
      </div>
      <div
        aria-checked={false}
        aria-selected={false}
        className="item"
        onClick={[Function]}
        option-value="option2"
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          Option 2
        </span>
      </div>
      <div
        aria-checked={false}
        aria-selected={false}
        className="item"
        onClick={[Function]}
        option-value="option3"
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          Option 3
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Elements/EditableLabel Basic 1`] = `
<label
  className=" editPlaceholder"
  onClick={[Function]}
>
  Click to edit
</label>
`;

exports[`Storyshots Elements/EditableLabel Editable 1`] = `
<label
  className=" editPlaceholder"
  onClick={[Function]}
>
  Enter your text here...
</label>
`;

exports[`Storyshots Elements/EditableLabel Read Only 1`] = `
<label
  className=" "
  onClick={[Function]}
>
  Sample Text
</label>
`;

exports[`Storyshots Elements/ErrorMessage Auto Hide 1`] = `
<div
  className="ui error message"
>
  <i
    aria-hidden="true"
    className="close icon"
    onClick={[Function]}
  />
  <div
    className="header"
  >
    Error Occured
  </div>
  <div
    className="content"
  >
    This error will disappear soon.
  </div>
</div>
`;

exports[`Storyshots Elements/ErrorMessage Basic 1`] = `
<div
  className="ui error message"
>
  <i
    aria-hidden="true"
    className="close icon"
    onClick={[Function]}
  />
  <div
    className="header"
  >
    Error Occured
  </div>
  <div
    className="content"
  >
    Invalid Blueprint ID provided
  </div>
</div>
`;

exports[`Storyshots Elements/ErrorMessage Custom Header 1`] = `
<div
  className="ui error message"
>
  <i
    aria-hidden="true"
    className="close icon"
    onClick={[Function]}
  />
  <div
    className="header"
  >
    Invalid input
  </div>
  <div
    className="content"
  >
    Please provide value according to blueprint naming rules.
  </div>
</div>
`;

exports[`Storyshots Elements/ErrorMessage Error Object 1`] = `
<div
  className="ui error message"
>
  <i
    aria-hidden="true"
    className="close icon"
    onClick={[Function]}
  />
  <div
    className="header"
  >
    Input error
  </div>
  <div
    className="content"
  >
    Please provide agent_private_key_path
  </div>
</div>
`;

exports[`Storyshots Elements/ErrorMessage Errors List 1`] = `
<div
  className="ui error message"
>
  <i
    aria-hidden="true"
    className="close icon"
    onClick={[Function]}
  />
  <div
    className="header"
  >
    Errors in the form
  </div>
  <ul
    className="list"
  >
    <li
      className="content"
    >
      Please provide deployment name
    </li>
    <li
      className="content"
    >
      Please provide agent_private_key_path
    </li>
  </ul>
</div>
`;

exports[`Storyshots Elements/Loading Basic 1`] = `
<div
  className="ui basic segment loadingSegment"
>
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="ui text loader"
      >
        Loading
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Elements/Loading Custom Message 1`] = `
<div
  className="ui basic segment loadingSegment"
>
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="ui text loader"
      >
        Loading files
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Elements/Menu Example 1`] = `
<div
  className="ui vertical menu"
>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="header"
    >
      Products
    </div>
    <div
      className="menu"
    >
      <a
        className="item"
        onClick={[Function]}
        option-value="enterprise"
      >
        Enterprise
      </a>
      <a
        className="item"
        onClick={[Function]}
        option-value="consumer"
      >
        Consumer
      </a>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="header"
    >
      CMS Solutions
    </div>
    <div
      className="menu"
    >
      <a
        className="item"
        onClick={[Function]}
        option-value="rails"
      >
        Rails
      </a>
      <a
        className="item"
        onClick={[Function]}
        option-value="python"
      >
        Python
      </a>
      <a
        className="item"
        onClick={[Function]}
        option-value="php"
      >
        Php
      </a>
    </div>
  </div>
</div>
`;

exports[`Storyshots Modal/Alert Basic 1`] = `
<div>
  <button
    className="ui button"
    onClick={[Function]}
  >
    Show Alert
  </button>
</div>
`;

exports[`Storyshots Modal/Confirm Basic 1`] = `
<div>
  <button
    className="ui button"
    onClick={[Function]}
  >
    Show Confirm
  </button>
</div>
`;

exports[`Storyshots Modal/ReadmeModal Basic 1`] = `
<div>
  <button
    className="ui button"
    onClick={[Function]}
  >
    Show Readme
  </button>
</div>
`;

exports[`Storyshots Popups/Popup Basic 1`] = `
<span
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Hover over me to see popup
</span>
`;

exports[`Storyshots Popups/PopupConfirm Basic 1`] = `
<button
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="button"
>
  Delete
</button>
`;

exports[`Storyshots Popups/PopupConfirm No Cancel Button 1`] = `
<button
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="button"
>
  Delete
</button>
`;

exports[`Storyshots Popups/PopupHelp Basic 1`] = `
<i
  aria-hidden="true"
  className="help circle icon"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;

exports[`Storyshots Popups/PopupHelp Custom Header 1`] = `
<i
  aria-hidden="true"
  className="help circle icon"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;

exports[`Storyshots Popups/PopupHelp Custom Trigger 1`] = `
<span
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Show Help
</span>
`;

exports[`Storyshots Popups/PopupMenu Basic 1`] = `
<i
  aria-hidden="true"
  className="content link icon"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;

exports[`Storyshots Popups/PopupMenu Custom Icon 1`] = `
<i
  aria-hidden="true"
  className="arrow down bordered link icon"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;

exports[`Storyshots Popups/PopupMenu Custom Position 1`] = `
<i
  aria-hidden="true"
  className="content link icon"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;

exports[`Storyshots Popups/PopupMenu Help Popup 1`] = `
<span
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <i
    aria-hidden="true"
    className="content link icon"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
</span>
`;
